### ISSUE TITLE
Doesn't work on Microsoft Teams 

### ISSUE THREAD

**Issue by sedveresk** [kyleneideck/BackgroundMusic#268]
MacOs High Sierra, not changing sound in Teams application

**Comment by pabclsn** [comment]
+1

**Comment by Anaxilaus** [comment]
Microsoft Teams app controls the notification sounds etc. and creates another thread for meetings called "Microsoft Teams Helper." You need to control this thread in the More Apps dropdown menu, marked as red below.
<img width="540" alt="Screen Shot 2020-03-27 at 3 33 09 PM" src="https://user-images.githubusercontent.com/24462323/77756575-80d6cc80-7040-11ea-96b7-33cc4ae8daf6.png">

**Comment by sedveresk** [comment]
ohhh, cool! thanks! is it similar with zoom? because does not seem to work either

**Comment by Anaxilaus** [comment]
No idea, haven't used zoom.

**Comment by pabclsn** [comment]
And same for Discord not working either ... :/

**Comment by dreamwhite** [comment]
![image](https://user-images.githubusercontent.com/17933708/77887079-3e90d380-726a-11ea-93fb-6ff69a26899e.png)

The one which has lower volume is for discord :)

**Comment by Anaxilaus** [comment]
Discord and Zoom also works. Extremely important to note, you should give access to microphone at `System Preferences -> Privacy & Security -> Microphone -> Background Music (Check)` to get Background Music working properly. Restart the applications and try again with the "Helper" method. This is an OS issue.

**Comment by Lyro1** [comment]
I can see the Helpers, but changing their volume does not change anything. I tried to restart Background Music and Teams, I verified that the Microphone permission was granted, and still not working...
<img width="540" alt="Capture d’écran 2020-04-20 à 14 40 37" src="https://user-images.githubusercontent.com/22308289/79752713-0898b980-8315-11ea-8176-cefcde5be0db.png">

**Comment by hemangshah** [comment]
> I can see the Helpers, but changing their volume does not change anything. I tried to restart Background Music and Teams, I verified that the Microphone permission was granted, and still not working...
> <img alt="Capture d’écran 2020-04-20 à 14 40 37" width="540" src="https://user-images.githubusercontent.com/22308289/79752713-0898b980-8315-11ea-8176-cefcde5be0db.png">

@kyleneideck I tried the same settings as already done by @Lyro1 too. But still it's same.

**Comment by kyleneideck** [comment]
@hemangshah I just tried it myself using the `/testcall` command in Microsoft Teams and I was able to control the volume using the slider for "Microsoft Teams H..." in More Apps. Are you sure Teams is playing its audio through the Background Music device?

Click the icon in the top right corner, pick Settings (or it might be Preferences or something), go to the Peripherals tab and make sure the output device is set to Background Music.

**Comment by Lyro1** [comment]
That was the issue. We need to make sure Teams is playing its audio through the Background Music device.

**Comment by hemangshah** [comment]
@kyleneideck @Lyro1 yeah, that was the issue, setting up within the MT app does the job ! 
Thank you so much.

**Comment by grouville** [comment]
Works for me when I am on speaker, however, when I am with my headphones, even after trying everything you said, it doesn't work

**Comment by kyleneideck** [comment]
@grouville Is it only Teams that doesn't work when you use headphones? Are they analog headphones that you're plugging into the built-in audio device?

**Comment by grouville** [comment]
@kyleneideck I just tried in VLC with my headphones plugged into the built-in audio device (Bose headphone); It works with VLC. I reinstalled Background Music with brew and restarted Teams. It now works.

The only difference I see is that I now have 2 "Microsoft Teams Helper" in the "More Apps" menu whereas the first time only one was showing up.
<img width="259" alt="image" src="https://user-images.githubusercontent.com/31691250/80484129-7f5a3600-8957-11ea-8961-174208a50c3d.png">

During the first install, I already was in a Teams conference, this may be the source of the issue

**Comment by surya-x** [comment]
> Microsoft Teams app controls the notification sounds etc. and creates another thread for meetings called "Microsoft Teams Helper." You need to control this thread in the More Apps dropdown menu, marked as red below.
> <img alt="Screen Shot 2020-03-27 at 3 33 09 PM" width="540" src="https://user-images.githubusercontent.com/24462323/77756575-80d6cc80-7040-11ea-96b7-33cc4ae8daf6.png">

This Worked for me! Thanks.

**Comment by yusronhanan** [comment]
> @kyleneideck I just tried in VLC with my headphones plugged into the built-in audio device (Bose headphone); It works with VLC. I reinstalled Background Music with brew and restarted Teams. It now works.
> 
> The only difference I see is that I now have 2 "Microsoft Teams Helper" in the "More Apps" menu whereas the first time only one was showing up.
> <img alt="image" width="259" src="https://user-images.githubusercontent.com/31691250/80484129-7f5a3600-8957-11ea-8961-174208a50c3d.png">
> 
> During the first install, I already was in a Teams conference, this may be the source of the issue -> the app couldn't find the other thread
> 
> Btw, thank you for this awesome app ! 🥇

This one works for me. Thanks.
